karapace/schema_reader.py

```diff
@@ -306,49 +300,52 @@ def handle_messages(self) -> None:
             if are_we_master is True:
                 watch_offsets = True
 
-        for _, msgs in raw_msgs.items():
-            schema_records_processed_keymode_canonical = 0
-            schema_records_processed_keymode_deprecated_karapace = 0
-            for msg in msgs:
+        schema_records_processed_keymode_canonical = 0
+        schema_records_processed_keymode_deprecated_karapace = 0
+        for msg in msgs:
+            try:
+                message_key = msg.key()
+                if message_key is None:
+                    continue
+                key = json_decode(message_key)
+            except JSONDecodeError:
+                LOG.exception("Invalid JSON in msg.key()")
```

Comment:
maybe logging also the message (truncated if its very big) could be helpful?

Reply:
I've added the offset to the log message, as you mentioned large keys (or values below) could be tricky. `LOG.exception` includes the error, that should already help with figuring out what went wrong.